### DISCUSSION TITLE
Search in ng-Tree by key

### DISCUSSION THREAD

**Discussion by MacherelR** [NG-ZORRO/ng-zorro-antd#8357] [Q&A]
Hi, 
I'm currently using the Searchable Tree (https://ng.ant.design/components/tree/en) which works like charm.
Only little question I asked myself, I'm using this to display an OPC-UA server tree structure, and for those who don't know about OPC-UA structures, each nodes are identified by a nodeId and a Name. In my case, I used the name as the "title" property of the NzTreeNode element and the nodeid as its key. But as I was asked, sometimes people would like to research based on the nodeId, therefore comes my question, is there a way to enable the research part of the Ng-Zorro Tree to search in elements' keys too ?

Best regards

Rémy

**Comment by ParsaArvanehPA** [comment]
Hi @MacherelR 
Have you tried using `[nzSearchFunc]`?
Apparently with it you can customize your tree's matching method([link](https://ng.ant.design/components/tree/en#api)).

Let me know if that doesn't do the job.
